src/CMakeLists.txt

```diff
@@ -357,29 +371,34 @@ elseif(${CMAKE_Fortran_COMPILER_ID} STREQUAL  GNU )
       # |             | -mno-fma4 -mavx2 -mfma                                            |      |
       # | GCC/openMPI | -O3 -march=znver3 -mtune=znver3 -ftree-vectorize -flto            | 280s | chatgpt recomendation
       # |             | -mcpu=znver3                                                      |      |
-      target_compile_options(${PROJECT_NAME} PRIVATE -march=znver3 -mtune=znver3 -ftree-vectorize -flto)      
+      if(CMAKE_BUILD_TYPE STREQUAL "Debug")
+         target_compile_options(${PROJECT_NAME} PRIVATE -march=znver3 -mtune=znver3)
+      else()
+         target_compile_options(${PROJECT_NAME} PRIVATE -march=znver3 -mtune=znver3 -ftree-vectorize -flto)
+      endif()      
    else()
       #[[if(NOT (${FESOM_PLATFORM_STRATEGY} STREQUAL ubuntu))
          target_compile_options(${PROJECT_NAME} PRIVATE -native)
       endif()  ]]  
    endif()
    
-   # # debugging related compiler flags
-   #target_compile_options(${PROJECT_NAME} PRIVATE 
-                           #-fallow-argument-mismatch 
-                           #-fno-fast-math -fno-signed-zeros 
-                           #-freciprocal-math 
-                           #-mveclibabi=svml 
-                           #-flto
-                           #-pg 
-                           #-fbacktrace 
-                           #-fcheck=all,bounds 
-                           #-finit-real=snan 
-                           #-finit-integer=-9999 
-                           #-fsanitize=undefined,address
-                        #)
-   ## if use -fsanitize=undefined,address you also need ... PRIVATE -lubsan)           
-   #target_link_libraries(${PROJECT_NAME} PRIVATE -lubsan)
+   # debugging related compiler flags (activated for Debug build type)
```

Comment:
we should use address sanitizer definitely for testing new code to check if we are allocating badly and creating leaks but I will skip -fsanitize for now as it need libasan to also be installed and present, this possibly need a cmake module to check this. so would skip this for now just, and ubsan seems to be a typo? should be -lasan or -static-libasan to compile options. More over any use of asan also need a env variable or someting to trace specific issue like ASAN_OPTIONS=detect_leaks=1 for detecting leaks. so this should be possibly done differently for it to be useful.